### ISSUE TITLE
Strange requirements for SLE rpm package

### ISSUE THREAD

**Issue by mkrsfcmp** [rsyslog/rsyslog#5106]
### Expected behavior

Only really essential requirements listed in the package.

### Actual behavior

rpm -Uvh rsyslog-8.2302.0-3.1.x86_64.rpm 
	group(news) is needed by rsyslog-8.2302.0-3.1.x86_64
	libfastjson-devel >= 0.99.8 is needed by rsyslog-8.2302.0-3.1.x86_64
	user(news) is needed by rsyslog-8.2302.0-3.1.x86_64

The *-devel packages should not be needed for running the package. Also the particular user/group requirement seems highly unusual.

### Steps to reproduce the behavior

rpm -Uvh rsyslog-8.2302.0-3.1.x86_64.rpm 

### Environment
- rsyslog version: 8.2302.0-3.1.x86_64.rpm - pulled from the SLE repo
- platform: SLES 15-SP3

**Comment by rgerhards** [comment]
thx - a patch would be appreciated

**Comment by mkrsfcmp** [comment]
Sure thing.
It's a patch against the rsyslog.spec file from the opensuse repository (I didn't find a spec anywhere here but on the other hand I didn't find a way to submit this there). 
I think the *-devel Requires was a simple typo. It was supposed to be BuildRequires. But the news user... IMHO it's a completely "outside" stuff that rsyslog package as such should not touch since it's not its realm. So I removed the news user related stuff from the postinstall script completely and also cut the requirement for the user existence.

```
--- rsyslog.spec.orig	2023-03-24 09:49:14.204178051 +0100
+++ rsyslog.spec	2023-03-24 09:51:30.847146084 +0100
@@ -188,7 +188,7 @@
 %define         rsyslog_module_dir_withdeps %{_libdir}/rsyslog/
 Url:            https://www.rsyslog.com/
 
-Requires:	libfastjson-devel >= 0.99.8
+BuildRequires:	libfastjson-devel >= 0.99.8
 
 %if 0%{?fedora} || 0%{?rhel_version} || 0%{?centos_version}
 BuildRequires: util-linux
@@ -366,11 +366,6 @@
 
 
 # PATCH-FIX-OPENSUSE rsyslog-unit.patch crrodriguez@opensuse.org Customize upstream systemd unit for openSUSE needs.
-%if 0%{?suse_version} > 1320
-  # Ensure "news" exists - bsc#1068678
-Requires(post): user(news)
-Requires(post): group(news)
-%endif
 
 # this is a dirty hack since % dir does only work for the specified directory and nothing above
 # but I want to be able to switch this to /etc/apparmor.d once the profiles received more testing
@@ -1179,15 +1174,6 @@
 	touch var/log/mail.info; chmod 640 var/log/mail.info
 	touch var/log/mail.warn; chmod 640 var/log/mail.warn
 	touch var/log/mail.err;  chmod 640 var/log/mail.err
-	test -f var/log/news && mv -f var/log/news var/log/news.bak
-	mkdir -p -m 0750 var/log/news
-	chown news:news  var/log/news
-	touch var/log/news/news.crit;   chmod 640 var/log/news/news.crit
-	chown news:news var/log/news/news.crit
-	touch var/log/news/news.err;    chmod 640 var/log/news/news.err
-	chown news:news var/log/news/news.err
-	touch var/log/news/news.notice; chmod 640 var/log/news/news.notice
-	chown news:news var/log/news/news.notice
 	#
 	# touch the additional log files we are using
 	#
```

**Comment by rgerhards** [comment]
Sorry for the very late reply. This may no longer be directly relevant to the original SLE package, but I am adding the answer for people who find this issue through search while we work through the backlog. We do not lightly close old issues and we do not close them just because of age.

The problematic requirements are not present in the current RPM packaging in this repository. `packaging/rpm/rpmbuild/SPECS/rsyslog-v8-stable.spec` uses `BuildRequires: libfastjson4-devel >= 0.99.8`, and I do not find the reported `Requires` on `libfastjson-devel`, `user(news)`, or `group(news)` in the current spec.

I am closing this as fixed/superseded in current packaging. If a current SLE package still has runtime requirements on development packages or unrelated users/groups, please open a new issue with the package version and `rpm -qpR` output.
